09-graphs-p2/01-graphs-review.md

```diff
@@ -0,0 +1,325 @@
+
+##### !end-explanation
+
+### !end-challenge
+
+<!-- ======================= END CHALLENGE ======================= -->
+
+<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->
+<!-- Replace everything in square brackets [] and remove brackets  -->
+
+### !challenge
+
+* type: multiple-choice
+* id: 2a316cff-6a03-4d90-9d15-4cf299afd223
+* title: Representing a graph with an adjacency dictionary
+* points: 0
+
+##### !question
+
+Which graph is represented by the following adjacency dictionary?
+
+```py
+adjacency_dict = {
+    "Seattle": ["Portland"],
+    "Portland": ["Dallas", "Seattle"],
+    "Detroit": ["Dallas"],
+    "Dallas": ["Boston", "Detroit", "Portland"],
+    "Boston": ["Dallas"]
+}
+```
+
+#### Option A
+![Graph A](images/graph_a.png)
+
+#### Option B
+![Graph B](images/graph_b.png)
+
+#### Option C
+![Graph C](images/graph_c.png)
+
+##### !end-question
+
+##### !options
+
+a| Option A
+b| Option B
+c| Option C
+
+##### !end-options
+
+##### !answer
+
+a|
+
+##### !end-answer
+
+##### !explanation
+
+Option A is correct because the nodes/edges in the adjacency dictionary correspond with the nodes and edges in the graph. 
+
+For instance:
+
+* Seattle is connected to only one city: Portland
+* Portland is connected to Dallas and Seattle
+* Dallas is connected to Boston, Detroit, and Portland
+* Detroit is connected to Dallas
+* Boston is connected to Dallas
+
+##### !end-explanation 
+
+### !end-challenge
+
+<!-- ======================= END CHALLENGE ======================= -->
+
+## Review of Breadth First Search & Depth First Search
+
+### Breadth First Search (BFS)
+You may recall from the last lesson that the Breadth First Search (BFS) algorithm starts with a particular node and then visits each node connected to the starting point before expanding outward.
+
+The BFS algorithm requires us to use a queue to visit nodes in a "level-by-level" ordering by adding each of the neighbors of the starting node to the queue. We then loop through the queue, adding unvisited neighbors to the list of visited nodes as well as to the queue to be further processed.
+
+### Depth First Search (DFS)
+You may also recall from the last lesson that the Depth First Search (DFS) algorithm starts at a node and then follows a path as deeply as possible before backing up and following the next path. We refer to this algorithm as a back-tracking algorithm.
+
+The DFS algorithm requires the use of a stack. The stack is used to further process nodes along the path, starting with the first node and then adding its neighbors to the stack before iterating on the nodes added to the stack. 
+
+### Why DFS does not find shortest path
+
+The main reason the DFS algorithm is not typically used to find the shortest path is because DFS is not a "greedy" algorithm. The algorithm is not designed to account for changes in logic based on the data it encounters, which is what is needed for finding the shortest path in a graph.
```

Comment:
This makes sense to me and I really like the callout, but I'm not sure students will have a strong enough grasp on this for this to give them a strong intuition since they haven't learned about different categories of algorithms yet. 

An explanation that has worked well for me in the past is to highlight that BFS chooses to visit nodes based on how close they are and we can leverage that to find the shortest path. DFS does not take nodes' proximity into consideration when making its traversal, so it's not useful for tracking the distance between nodes.